File: Docs/UT7-AndroidApplicationDevelopment/Example03/app/src/main/java/com/raks/intents/implicit/MainActivity.java
package com.raks.intents.implicit;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.inputmethod.EditorInfo;
import android.widget.EditText;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    private static final String GEO_0_0_Q_S = "geo:0,0?q=%s";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        final EditText input = findViewById(R.id.address);

        input.setOnEditorActionListener((textView, i, keyEvent) -> {
            if (i == EditorInfo.IME_ACTION_DONE) {
                String address   = Uri.encode(textView.getText().toString());
                Intent mapIntent = new Intent(Intent.ACTION_VIEW);
                Uri    uri       = Uri.parse(String.format(GEO_0_0_Q_S, address));

                mapIntent.setData(uri);
                startActivity(mapIntent);
                return true;
            }
            return false;
        });
    }

}

File: Projects/P8-IntentsImplicitos/app/src/main/kotlin/com/raks/android/MainActivity.kt
package com.raks.android

import android.content.Intent
import android.net.Uri
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val btn = findViewById<Button>(R.id.button)
        val url = findViewById<EditText>(R.id.edit_text)
        btn.setOnClickListener { openWebPage(url.text.toString()) }
    }

    private fun openWebPage(url: String) {
        val intent = Intent(Intent.ACTION_VIEW, Uri.parse(url))
        startActivity(intent)
    }

}